### PR TITLE
[Tests]Remove migration test skips

### PR DESCRIPTION
**Pull request by Barakmor1**
Removed explicit skipping from test code. Added documentation for skipping tests based on cluster requirements using the `KUBEVIRT_E2E_SKIP` variable. Also added a decorator for skipping tests requiring two schedulable nodes, which is useful for single-node clusters.

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by dankenigsberg** [comment]
I think this deserves a release note: how would you now test KubeVirt on a single-node k8s? We should document somewhere how a tester can deselect migration tests.

**Comment by EdDev** [comment]
> I think this deserves a release note: how would you now test KubeVirt on a single-node k8s? We should document somewhere how a tester can deselect migration tests.

+1
The PR will probably not be accepted without a detailed explanation on how existing executions of the e2e test against a cluster with only a single worker node is expected to work.

**Comment by Barakmor1** [comment]
Hi, thank you for the note! As far as I know, we typically don’t include testing-related details in the release notes. There are several tests that won't work in certain environments. 
Maybe it would be helpful to provide clear instructions on how and when tests should be run in the user guide instead. I’m not sure that release notes are the best place for this kind of information.

WDYT?

**Comment by Barakmor1** [comment]
i'll add the `RequiresTwoSchedulableNodes` decorator to these tests for now.

**Comment by EdDev** [comment]
> Hi, thank you for the note! As far as I know, we typically don’t include testing-related details in the release notes. There are several tests that won't work in certain environments. Maybe it would be helpful to provide clear instructions on how and when tests should be run in the user guide instead. I’m not sure that release notes are the best place for this kind of information.
> 
> WDYT?

Lets start in documenting the alternative and implication in the PR and commit description.
What happens now, what will happen after and how to mitigate it if one runs against a single node.

Then, you can add the same info in the e2e documentation under `/tests`.

**Comment by Barakmor1** [comment]
@EdDev PTAL

**Review by dankenigsberg — APPROVED** [review]
The order of the commits does not make sense to me: you are removing the skips before you add the decorator. But I'm not be holding this positive PR on this.

File: tests/README.md [hunk 1]
```diff
@@ -117,3 +117,29 @@ docker run \
         --deploy-testing-infra \
         --path-to-testing-infra-manifests=data/manifests
 ```
+
+
+## Skipping Tests When Cluster Requirements Aren't Met
```
Comment (iholder101) [inline tests/README.md:122]:
```suggestion
## Skipping tests when cluster requirements aren't met
```

File: tests/README.md [hunk 2]
```diff
@@ -117,3 +117,29 @@ docker run \
...
+KubeVirt provides [tests decorators](https://github.com/kubevirt/kubevirt/blob/main/tests/decorators/decorators.go) 
+to indicate the cluster requirements for certain tests. If your 
+cluster doesn't meet these requirements, you can choose to skip 
+the relevant tests by explicitly setting the `KUBEVIRT_E2E_SKIP`
```
Comment (iholder101) [inline tests/README.md:126]: Can you also briefly mention `KUBEVIRT_E2E_FOCUS` and `KUBEVIRT_LABEL_FILTER`?

File: tests/README.md [hunk 3]
```diff
@@ -117,3 +117,29 @@ docker run \
...
+export KUBEVIRT_E2E_SKIP=requires-two-schedulable-nodes
+```
+
+You can skip multiple tests by combining decorators with the | symbol. 
+For example, to skip tests requiring two schedulable nodes and 
+specific storage, run:
+
+``` bash
+export KUBEVIRT_E2E_SKIP=requires-two-schedulable-nodes|storage-req
+```
```
Comment (iholder101) [inline tests/README.md:142]: I think you can generalize this by mentioning that this is a regex [1], and keep the current use as an example.

[1] https://onsi.github.io/ginkgo/#description-based-filtering

**Review by iholder101 — COMMENTED** [review]
Left some tiny nits, but looks great in general.
Thank you @Barakmor1!

File: tests/README.md (hunk 1 above)
Comment (Barakmor1) [reply tests/README.md]: Done

File: tests/README.md (hunk 2 above)
Comment (Barakmor1) [reply tests/README.md:125]: Done

File: tests/README.md (hunk 3 above)
Comment (Barakmor1) [reply tests/README.md]: Done

**Comment by iholder101** [comment]
Thank you @Barakmor1!

/approve

**Comment by Barakmor1** [comment]
@dankenigsberg PTAL

**Review by dankenigsberg — APPROVED** [review]

**Pull request merged**